Automattic/newspack-listings
# fix: prevent dupes of self-serve products; manual creation and deletion

### All Submissions:






### Changes proposed in this Pull Request:



When self-serve listings are enabled, the plugin automatically generates the required WooCommerce products it needs to enable the reader-facing purchase flow. However, if any of these products is unpublished or deleted, the plugin will regenerate them, potentially resulting in many unwanted duplicates.

This PR removes the auto-generation and instead requires an intentional action on the part of site admins to create the products before self-serve features can be used. It also allows for deletion of the products (thus disabling self-serve purchase features) if self-serve isn't needed on the site.

The UI is fairly bare-bones as self-serve features are still experimental, but this does prevent a situation where unwanted products can be created en masse.

### How to test the changes in this Pull Request:

1. On a new site, install this plugin, WooCommerce, and WooCommerce Subscriptions. Add the `define( 'NEWSPACK_LISTINGS_SELF_SERVE_ENABLED', true );` flag to `wp-config.php` to enable self-serve features.
2. Go to **Listings > Settings** and observe that the "Self-Serve Listings" section is available, but its fields are disabled.

<img width="667" alt="Screen Shot 2022-07-12 at 11 17 19 AM" src="https://user-images.githubusercontent.com/2230142/178553316-fad3b570-9d9c-4782-ac2f-8fddc6e9940a.png">

3. Edit a post or page and attempt to insert a Self-Serve purchase form block, and confirm that you can't.
4. Go to Products and confirm that no self-serve listings products exist yet, as you haven't created them.
5. Go back to Listings > Settings and click the "Click here to enable self-serve listings features on this site" link.
6. Confirm that the self-serve settings are now editable.
7. Go to Products and confirm that the self-serve products were generated.
8. Edit a post or page and confirm that you can insert a Self-Serve purchase form block, and that it functions as expected on the front-end.
9. Go back to Listings > Settings and click on the "Click here to disable self-serve listings features on this site" link.
10. Confirm that the self-serve settings are disabled again, the products have been deleted, and that the Self-Serve purchase form block is no longer available in the editor or on the front-end.

### Other information:
